packages/expo-errors/android/src/main/java/expo/errors/HostApplicationCausedPlatformException.java

```diff
@@ -0,0 +1,25 @@
+package expo.errors;
+
+/**
+ * Helper class to make errors easier to handle. The native promise.reject
+ * method on each platform should know about this class and be able to get
+ * the code itself when passed an instance of this class.
+ */
+
+public abstract class HostApplicationCausedPlatformException extends RuntimeException {
```

Comment:
It looks like there are two parts to this error -- the code and the "HostApplicationCaused" part.

1) It seems like we want a way to know that we can get a code from an error so that the bridge can send the code back to JS. 

2) I'm not sure what "HostApplication" refers to (the Java code?) but presumably we want to signal whether the native app is in a good enough state that reloading the JS is OK (c.f. JSApplicationCausedNativeException).

These two things are separate in that we might have an error that's not the developer's fault (e.g. tried to read from network but there's no connection) but still include a code like `ERR_NETWORK_OFFLINE`.